test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -552,17 +591,29 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 		}
 		switch concreteOp := realOp.(type) {
 		case *createNodesOp:
-			nodePreparer, err := getNodePreparer(fmt.Sprintf("node-%d-", opIndex), concreteOp, client)
+			nodeManager, err := getNodeManager(fmt.Sprintf("node-%d-", opIndex), concreteOp, client)
 			if err != nil {
 				b.Fatalf("op %d: %v", opIndex, err)
 			}
-			if err := nodePreparer.PrepareNodes(nextNodeIndex); err != nil {
+			if err := nodeManager.PrepareNodes(nextNodeIndex); err != nil {
 				b.Fatalf("op %d: %v", opIndex, err)
 			}
 			b.Cleanup(func() {
-				nodePreparer.CleanupNodes()
+				nodeManager.CleanupNodes()
 			})
 			nextNodeIndex += concreteOp.Count
+			if concreteOp.Name != nil {
+				nodeManagers[*concreteOp.Name] = nodeManager
+			}
+
+		case *deleteNodesOp:
+			nodeManager, ok := nodeManagers[concreteOp.Name]
+			if !ok {
+				b.Fatalf("op %d: cannot get node manager. Make sure that the name, %s, is correct", opIndex, concreteOp.Name)
+			}
+			if err := nodeManager.DeleteNodes(concreteOp.Count); err != nil {
```

Comment:
So this is my first time looking at this code and in general it looks OK but was hoping you could help educate me a bit.

So it seems there are numerous nodenamanagers? I would have originally thought there is a single node manager that manages all the nodes. You see to have M managers and N nodes for each manager, is that right? Why is that?

Here you used the node name to lookup the manager, but then delete just the first `count` nodes. If it is by name wouldnt you need to iterate through the list/identify the exact node? Or is the manager really guaranteed to have just one node, so looking up the manager and deleting 1 is sure to hit the node you intend"

Reply:
Hello @johnSchnake. Sorry for toooo late reply. 

I describe what this PR does.

On scheduler_perf, users cannot specify node names when they create nodes. nodes are created with GenerateName and added random suffix. And, nodes created in a single operation has the same specification, just a different name

And I add Name field on createNodeOp (on [L205](https://github.com/kubernetes/kubernetes/pull/101854/files#diff-79447214c2082cd5c13491439ad62ada796aa79ca2c436b35eafd4c0ec0861a7R205)) so that users can specify nodes to be deleted when deleting a node.

When we delete nodes, users specify createNodeOp.Name created nodes they want to delete. And we only have to think about which createNodeOp created nodes. It doesn't matter which of nodes created by the operation we delete because all nodes created by the same createNodeOp have the same spec.

---

So, reply to your questions.

> You see to have M managers and N nodes for each manager, is that right? Why is that?

It's right. One nodeManager has nodes that one createNodeOp created. And nodeManagers has many nodeManager and use createNodeOp.Name as the key.

>  If it is by name wouldnt you need to iterate through the list/identify the exact node? Or is the manager really guaranteed to have just one node, so looking up the manager and deleting 1 is sure to hit the node you intend"

We only have to think about which nodeManager has nodes users want to delete. As described above, it doesn't matter which of nodes created by the same operation we delete.